functions/venue.js

```diff
@@ -553,26 +569,15 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
     updated.radioStations = [data.radioStations];
   }
 
+  // @debt this would currently allow any value to be set in this field, not just booleans
   updated.requiresDateOfBirth = data.requiresDateOfBirth || false;
 
-  switch (updated.template) {
-    case VenueTemplate.jazzbar:
-    case VenueTemplate.performancevenue:
-    case VenueTemplate.artpiece:
-    case VenueTemplate.audience:
-      if (data.iframeUrl) {
-        updated.iframeUrl = data.iframeUrl;
-      }
-      break;
-    case VenueTemplate.zoomroom:
-    case VenueTemplate.artcar:
-      if (data.zoomUrl) {
-        updated.zoomUrl = data.zoomUrl;
-      }
-      break;
+  if (IFRAME_TEMPLATES.includes(updated.template) && data.iframeUrl) {
+    updated.iframeUrl = data.iframeUrl;
+  }
 
-    default:
-      break;
+  if (ZOOM_URL_TEMPLATES.includes(updated.template) && data.zoomUrl) {
+    updated.zoomUrl = data.zoomUrl;
   }
```

Comment:
RE: https://github.com/sparkletown/sparkle/pull/1274#pullrequestreview-613992793

> I couldn't spot the root of the issue

@mike-lvov The root of the issue was that `iframeUrl` is never set for `VenueTemplate.embedded` / `VenueTemplate.firebarrel` here in `exports.updateVenue`

![image](https://user-images.githubusercontent.com/753891/111432136-be9c9b80-8750-11eb-9dd1-4f6de0768821.png)